shell/platform/embedder/tests/embedder_test_backingstore_producer_metal.mm

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/tests/embedder_test_backingstore_producer_metal.h"
+
+#include "flutter/fml/logging.h"
+#include "third_party/skia/include/core/SkColorSpace.h"
+#include "third_party/skia/include/gpu/ganesh/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/ganesh/SkSurfaceGanesh.h"
+#include "third_party/skia/include/gpu/ganesh/mtl/GrMtlBackendSurface.h"
+#include "third_party/skia/include/gpu/ganesh/mtl/GrMtlTypes.h"
+
+namespace flutter::testing {
+
+EmbedderTestBackingStoreProducerMetal::EmbedderTestBackingStoreProducerMetal(
+    sk_sp<GrDirectContext> context,
+    RenderTargetType type)
+    : EmbedderTestBackingStoreProducer(context, type),
+      test_metal_context_(std::make_unique<TestMetalContext>()) {}
+
+EmbedderTestBackingStoreProducerMetal::~EmbedderTestBackingStoreProducerMetal() = default;
+
+bool EmbedderTestBackingStoreProducerMetal::Create(const FlutterBackingStoreConfig* config,
+                                                   FlutterBackingStore* backing_store_out) {
+  // TODO(gw280): Use SkSurfaces::RenderTarget instead of generating our
+  // own MTLTexture and wrapping it.
+  auto surface_size = SkISize::Make(config->size.width, config->size.height);
+  auto texture_info = test_metal_context_->CreateMetalTexture(surface_size);
+
+  GrMtlTextureInfo skia_texture_info;
+  skia_texture_info.fTexture.reset(SkCFSafeRetain(texture_info.texture));
+  GrBackendTexture backend_texture = GrBackendTextures::MakeMtl(
```

Comment:
Nothing to do here, but all these tests/test harness that use Skia types directly like this will need to either be rewritten or deleted next year

Reply:
Yup. I suspect there's a non-zero amount of Skia-specific code in the `embedder_unittests` source too for that matter.

Refactoring all this stuff out into backend-specific classes hopefully makes either of those options easier.